prow/cmd/generic-autobumper/bumper/bumper.go

```diff
@@ -371,19 +371,52 @@ func Run(o *Options) error {
 			return fmt.Errorf("failed to create the PR: %w", err)
 		}
 	} else {
-		changeId, err := getChangeId(o.Gerrit.Author, o.Gerrit.AutobumpPRIdentifier)
+		if err := Call(stdout, stderr, gitCmd, "config", "http.cookiefile", o.Gerrit.CookieFile); err != nil {
+			return fmt.Errorf("unable to load cookiefile: %v", err)
+		}
+		if err := Call(stdout, stderr, gitCmd, "config", "user.name", o.Gerrit.Author); err != nil {
+			return fmt.Errorf("unable to set username: %v", err)
+		}
+		if err := Call(stdout, stderr, gitCmd, "config", "user.email", o.Gerrit.Email); err != nil {
+			return fmt.Errorf("unable to set password: %v", err)
+		}
+		if err := Call(stdout, stderr, gitCmd, "remote", "add", "upstream", o.Gerrit.HostRepo); err != nil {
+			return fmt.Errorf("unable to add upstream remote: %v", err)
+		}
+
+		changeId, err := getChangeId(o.Gerrit.Author, o.Gerrit.AutobumpPRIdentifier, "")
 		if err != nil {
 			return fmt.Errorf("Failed to create CR: %w", err)
 		}
-		msg := makeGerritCommit(o.Prefixes, versions, o.Gerrit.AutobumpPRIdentifier, changeId)
-		// TODO(mpherman): Add reviewers to CreateCR
-		if err := createCR(msg, "master", changeId, o.Gerrit.HostRepo, o.Gerrit.CookieFile, o.Gerrit.Author, o.Gerrit.Email, nil, nil, stdout, stderr); err != nil {
-			return fmt.Errorf("Failled to create the CR: %w", err)
+
+		err = gerritCommitandPush(o.Prefixes, versions, o.Gerrit.AutobumpPRIdentifier, changeId, nil, nil, stdout, stderr)
+		// If failed to push because a closed PR already exists with this change ID (the PR was abandoned). Hash the ID again and try one more time.
+		if err != nil && strings.Contains(err.Error(), "failed to push some refs") && strings.Contains(err.Error(), "closed") {
+			logrus.Warn("Error pushing CR due to already used ChangeID. PR may have been abandoned. Trying again with new ChangeID.")
+			changeId, err := getChangeId(o.Gerrit.Author, o.Gerrit.AutobumpPRIdentifier, changeId)
+			if err != nil {
+				return err
+
+			}
+			err = gerritCommitandPush(o.Prefixes, versions, o.Gerrit.AutobumpPRIdentifier, changeId, nil, nil, stdout, stderr)
```

Comment:
Please make sure this doesn't create a new commit

Reply:
Did not think of this. Added a call to `git reset HEAD^`before the second commit.

Reply:
hmm... wouldn't that flush away the changes as well?

Reply:
I believe this un-stages the changes, but does not revert the changes. I tested locally to be sure